packages/react/src/profile/useProfile.ts

```diff
@@ -20,7 +20,12 @@ export type UseProfileByHandleArgs = {
 };
 
 export type UseProfileArgs = WithObserverIdOverride<
-  XOR<UseProfileByIdArgs, UseProfileByHandleArgs>
+  XOR<
+    {
+      skip: true;
```

Comment:
Do we need this? I think it's part of a wider conversation we need to have on how and if we need to have conditionally triggerable queries (either via a `skip` of by a laxy-query approach).

But I do like your take using `XOR`, it does address my reservations on type safety as one cannot provide data until it's skipped.

Reply:
In this PR I would still use conditional rendering to render downstream hooks. Then we can open a separate conversation about this `skip` approach, which I reckon could be done across the board to all read hooks.

Reply:
the problem is that I'm fetching xmtp conversation(s), extracting `profileIds` from the conversations (if lens convos), and then fetching full profiles with `useProfile(s)` hooks. It is possible that no conversation is a lens conversation, which means I'm calling `useProfile(s)` without any profileId, which returns an error from the lens API. I would like to not trigger `useProfile(s)` hooks if no profileIds found in conversations. any ideas how to do it?

Reply:
Seems a good opportunity to use `OneOf` (https://timhwang21.gitbook.io/index/programming/typescript/xor-type) generalization of `XOR`.

Can we flag the `{ skip: true }` option as `@experimental`?